esmvalcore/preprocessor/_regrid_iris_esmf_regrid.py

```diff
@@ -0,0 +1,208 @@
+        :mod:`esmpy` methods
+        :attr:`~esmpy.api.constants.RegridMethod.CONSERVE`,
+        :attr:`~esmpy.api.constants.RegridMethod.BILINEAR` or
+        :attr:`~esmpy.api.constants.RegridMethod.NEAREST_STOD` used to
+        calculate regridding weights.
+    mdtol:
+        Tolerance of missing data. The value returned in each element of
+        the returned array will be masked if the fraction of masked data
+        exceeds ``mdtol``. ``mdtol=0`` means no missing data is tolerated while
+        ``mdtol=1`` will mean the resulting element will be masked if and only
+        if all the contributing elements of data are masked. If no value is
+        given, this will default to 1 for conservative regridding and 0
+        otherwise. Only available for methods 'bilinear' and 'conservative'.
+    use_src_mask:
+        If True, derive a mask from (first time step) of the source cube,
+        which will tell :mod:`esmpy` which points to ignore. If an array is
+        provided, that will be used.
+        If set to :obj:`None`, it will be set to :obj:`True` for methods
+        'bilinear' and `conservative' and to :obj:`False` for method 'nearest'.
+    use_tgt_mask:
+        If True, derive a mask from (first time step) of the target cube,
+        which will tell :mod:`esmpy` which points to ignore. If an array is
+        provided, that will be used.
+        If set to :obj:`None`, it will be set to :obj:`True` for methods
+        `bilinear' and 'conservative' and to :obj:`False` for method 'nearest'.
+    src_resolution:
+        If present, represents the amount of latitude slices per source cell
+        given to ESMF for calculation. If resolution is set, the source cube
+        must have strictly increasing bounds (bounds may be transposed
+        plus or minus 360 degrees to make the bounds strictly increasing).
+        Only available for method 'conservative'.
+    tgt_resolution:
+        If present, represents the amount of latitude slices per target cell
+        given to ESMF for calculation. If resolution is set, the target cube
+        must have strictly increasing bounds (bounds may be transposed
+        plus or minus 360 degrees to make the bounds strictly increasing).
+        Only available for method 'conservative'.
+    tgt_location:
+        Either "face" or "node". Describes the location for data on the mesh
+        if the target is not a :class:`~iris.cube.Cube`.
+
+    Attributes
+    ----------
+    kwargs:
+        Keyword arguments that will be provided to the regridder.
+    """
+
+    def __init__(
+        self,
+        method: Literal["bilinear", "conservative", "nearest"],
+        mdtol: float | None = None,
+        use_src_mask: bool | np.ndarray = True,
+        use_tgt_mask: bool | np.ndarray = True,
+        src_resolution: int | None = None,
+        tgt_resolution: int | None = None,
+        tgt_location: Literal['face', 'node'] | None = None,
```

Comment:
Would it maybe make sense to pass all of these as `**kwargs`? It's nice to have them explicitly documented here, but if the regridders support new arguments in the future or change arguments we will need to adapt this code here to support this.